src/sql/workbench/parts/query/browser/query.contribution.ts

```diff
@@ -404,6 +404,121 @@ let registryProperties = {
 		'type': 'boolean',
 		'default': false,
 		'description': localize('mssql.intelliSense.lowerCaseSuggestions', 'Should IntelliSense suggestions be lowercase')
+	},
+	'mssql.query.rowCount': {
+		'type': 'number',
```

Comment:
Are these going to be exposed anywhere so callers can discover the available options for a provider?

Reply:
Yes, these are exposed in the Setting Editor UI with descriptions.

Reply:
I was more coming from the standpoint of an extension developer. So in your example above how would an extension developer know what valid options there are (and more specifically what key name to use for each of them) without digging through the code? I'm thinking of something like this :

```
	enum QueryOption {
		'includeActualExecutionPlanXml',
		'ansiNulls'
	}
	let options: Map<QueryOption | string, any> = new Map<QueryOption | string, any>();
	options[QueryOption.ansiNulls] = true;
	options['customOption'] = 'foobar';
```

This would let there be a set of known options users could specify in addition to providing the ability to use any string (for other providers for example). 

I guess we could even just leave the Map as taking in a string and just have the enum be available so that callers can use it to get the known names but not have this implementation tied to the MS SQL options.

Reply:
I like this idea.  Let me see how that works.  Otherwise, I might include that in the next PR which will a few bugs in the query editor extensibility APIs (since this PR has already been tested and is basically ready to go).

Reply:
So the challenge with this is that those options are MSSQL-provider specific and we're trying to keep provider specific details out of azdata.  Probably it makes sense to have the MSSQL extension publish it's own `.d.ts` file with those definitions.  I'll look at that separately when I put together my test extension.

Reply:
Yeah I figured something like that would be an issue. And I can't think of a good way to enforce that so going the route of just having each provider define their own options in a typings file seems to make the most sense for now.